File: planner-todo/src/main/java/ru/javabegin/micro/planner/todo/service/CategoryService.java
package ru.javabegin.micro.planner.todo.service;


import org.springframework.stereotype.Service;
import ru.javabegin.micro.planner.entity.Category;
import ru.javabegin.micro.planner.todo.repo.CategoryRepository;

import javax.transaction.Transactional;
import java.util.List;

@Service


@Transactional
public class CategoryService {

    private final CategoryRepository repository;

    public CategoryService(CategoryRepository repository) {
        this.repository = repository;
    }

    public Category findById(long id) {
        return repository.findById(id).get();
    }

    public List<Category> findAll(Long userId) {
        return repository.findByUserIdOrderByTitleAsc(userId);
    }

    public Category add(Category category) {
        return repository.save(category);
    }

    public Category update(Category category) {
        return repository.save(category);
    }

    public void deleteById(long id) {
        repository.deleteById(id);
    }

    public List<Category> findByTitle(String title, Long userId) {
        return repository.findByTitle(title, userId);
    }
}

File: planner-todo/src/main/java/ru/javabegin/micro/planner/todo/service/CategoryService.kt
package ru.javabegin.micro.planner.todo.service

import org.springframework.stereotype.Service
import ru.javabegin.micro.planner.entity.Category
import ru.javabegin.micro.planner.todo.repo.CategoryRepository
import javax.transaction.Transactional

@Service
@Transactional
class CategoryService(private val repository: CategoryRepository) {
    fun findById(id: Long): Category {
        return repository.findById(id).get()
    }

    fun findAll(userId: Long): List<Category> {
        return repository.findByUserIdOrderByTitleAsc(userId)
    }

    fun add(category: Category): Category {
        return repository.save(category)
    }

    fun update(category: Category): Category {
        return repository.save(category)
    }

    fun deleteById(id: Long) {
        repository.deleteById(id)
    }

    fun findByTitle(title: String?, userId: Long): List<Category> {
        return repository.findByTitle(title, userId)
    }
}